crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -3619,7 +3619,7 @@ where
             ExprKind::Lambda { .. } => {
                 self.deferred
                     .lambdas
-                    .push((expr, (self.ctx.scope_id, self.ctx.parents.clone())));
+                    .push((expr, (self.ctx.scope_id, self.ctx.node_id)));
```

Comment:
Does `node_id` match the id of the current stmt? If so, how about `self.ctx.current_stmt_id()`? It would clarify the relationship to `current_stmt`